src/Facade/Services/StoresPalletFacadeService.cs

```diff
@@ -0,0 +1,178 @@
+﻿namespace Linn.Stores2.Facade.Services
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq.Expressions;
+    using System.Threading.Tasks;
+    using Amazon.SimpleEmail.Model;
+
+    using Linn.Common.Facade;
+    using Linn.Common.Persistence;
+    using Linn.Stores2.Domain.LinnApps;
+    using Linn.Stores2.Domain.LinnApps.Stock;
+    using Linn.Stores2.Facade.Common;
+    using Linn.Stores2.Resources;
+
+    public class StoresPalletFacadeService : AsyncFacadeService<StoresPallet, int, StoresPalletResource, StoresPalletResource, StoresPalletResource>
+    {
+        private readonly IRepository<StockPool, string> stockPoolRepository;
+
+        private readonly IRepository<LocationType, string> locationTypeRepository;
+
+        private readonly IRepository<StorageLocation, int> storageLocationTypeRepository;
+
+        private readonly IRepository<StoresPallet, int> palletRepository;
+        public StoresPalletFacadeService(
+            IRepository<StoresPallet, int> repository,
+            ITransactionManager transactionManager,
+            IBuilder<StoresPallet> resourceBuilder,
+            IRepository<StockPool, string> stockPoolRepository,
+            IRepository<LocationType, string> locationTypeRepository,
+            IRepository<StorageLocation, int> storageLocationTypeRepository)
+            : base(repository, transactionManager, resourceBuilder)
+        {
+            this.stockPoolRepository = stockPoolRepository;
+            this.palletRepository = repository;
+            this.locationTypeRepository = locationTypeRepository;
+            this.storageLocationTypeRepository = storageLocationTypeRepository;
+        }
+
+        protected override async Task<StoresPallet> CreateFromResourceAsync(
+            StoresPalletResource resource,
+            IEnumerable<string> privileges = null)
+        {
+            var alreadyExist = this.palletRepository.FindById(resource.PalletNumber);
+
+            if (alreadyExist != null)
+            {
+                throw new AlreadyExistsException($"Pallet {resource.PalletNumber} already exists.");
+            }
+
+            var stockPool = this.stockPoolRepository.FindById(resource.DefaultStockPoolId);
+
+            if (stockPool == null && resource.DefaultStockPoolId != null)
+            {
+                throw new NullReferenceException($"Stock pool {resource.DefaultStockPool.StockPoolCode} not found.");
+            }
```

Comment:
is this a domain rule? i.e. 'Pallets must have a stock pool'? If so move this code into the domain and unit test